lib/internal/http2/compat.js

```diff
@@ -349,24 +374,34 @@ class Http2ServerRequest extends Readable {
       return;
     this[kStream].setTimeout(msecs, callback);
   }
-
-  [kFinish]() {
-    const state = this[kState];
-    if (state.closed)
-      return;
-    state.closed = true;
-    this.push(null);
-    this[kStream][kRequest] = undefined;
-    // if the user didn't interact with incoming data and didn't pipe it,
-    // dump it for compatibility with http1
-    if (!state.didRead && !this._readableState.resumeScheduled)
-      this.resume();
-    this.emit('close');
-  }
 }
 
 function onStreamTrailersReady() {
-  this[kStream].sendTrailers(this[kTrailers]);
+  const stream = this;
+  const res = stream[kResponse];
```

Comment:
Why these extra variables instead of just:

```js
this.sendTrailers(this[kResponse][kTrailers]);
```

?

Reply:
No particular reason. Just style. I'm fine with changed if you want me to?